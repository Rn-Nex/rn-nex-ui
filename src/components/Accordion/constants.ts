export const ACCORDION_DETAILS_OPACITY_DURATION = 200;
export const ACCORDION_DETAILS_DEFAULT_EXPANDED = false;
export const ROTATE_ANIMATION_DURATION = 150;
export const HEIGHT_VALUE_ANIMATION_DURATION = 200;
export const ROTATE_ANIMATION_RANGE = ['0deg', '180deg'];
