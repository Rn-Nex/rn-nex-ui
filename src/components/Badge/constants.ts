export const BADGE_TOP_POSITION = -15;
export const BADGE_LEFT_POSITION = 10;
export const BADGE_ANIMATION_DURATION = 100;
export const BADGE_MAX_DEFAULT_VALUE = 100;
export const BADGE_DEFAULT_WIDTH = 10;
export const BADGE_DEFAULT_HEIGHT = 10;
