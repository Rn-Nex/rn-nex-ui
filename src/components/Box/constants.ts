export enum BOX_GENERATED_CLASSNAMES {
  RN_NIX_BOX_SX_CLASS = 'Rn-box-generated-sx-class',
  RN_NIX_CONTAINER_SX_CLASS = 'Rn-container-generated-sx-class',
  RN_NIX_CONTAINER_INNER_SX_CLASS = 'Rn-container-generated-inner-sx-class',
}

export const XS_MAX_WIDTH = '40%';
export const SM_MAX_WIDTH = '50%';
export const MD_MAX_WIDTH = '70%';
export const LG_MAX_WIDTH = '80%';
export const xl_MAX_WIDTH = '90%';
