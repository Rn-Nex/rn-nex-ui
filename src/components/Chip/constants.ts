export const ADORNMENT_WRAPPER_SPACE = 20;
