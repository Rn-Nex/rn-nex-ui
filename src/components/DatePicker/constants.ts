export const WRAPPER_DEFAULT_HEIGHT = 100;
export const WRAPPER_BOTTOM_OFFSET = 50;
