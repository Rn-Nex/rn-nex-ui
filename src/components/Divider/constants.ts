export const DEFAULT_WIDTH = '50%';
export const DEFAULT_HEIGHT = 1;
export const MIDDLE_WIDTH = '42%';
export const MIDDLE_VARIATION_SIDE_SPACE = 40;
export const MIDDLE_VARIATION_HEIGHT_SPACE = 20;
