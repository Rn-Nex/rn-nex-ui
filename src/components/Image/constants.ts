export const IMAGE_ROUNDED_RADIUS = 100;
export const IMAGE_ROUNDED_SM_RADIUS = 20;
export const IMAGE_ROUNDED_MD_RADIUS = 30;
export const IMAGE_ROUNDED_LG_RADIUS = 40;
export const IMAGE_ROUNDED_XL_RADIUS = 60;
export const IMAGE_SQUARE_RADIUS = 0;
