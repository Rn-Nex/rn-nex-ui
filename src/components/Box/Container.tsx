import React, { useMemo } from 'react';
import { StyleSheet, View } from 'react-native';
import { generateElementStyles } from '../../utils';
import { containerStyles, generateContainerStyles } from './Box.style';
import { ContainerProps } from './Box.types';
import { BOX_CLASSNAMES } from './constants';

export const Container = React.forwardRef<View, ContainerProps>(
  ({ sx, style, maxWidth, disableGutters, children, containerStyles: containerWrapperStyles, testID, ...props }, ref) => {
    const containerWrapperTestId = testID ? `${testID}_outer` : undefined;

    const containerGeneratedStyles = useMemo(() => {
      return StyleSheet.create({
        [BOX_CLASSNAMES.RN_NIX_CONTAINER_INNER_SX_CLASS]: {
          ...generateContainerStyles({ maxWidth, disableGutters }),
          ...generateElementStyles(sx || {}),
        },
      });
    }, [sx, maxWidth, disableGutters]);

    return (
      <View
        style={[containerStyles[BOX_CLASSNAMES.RN_NIX_CONTAINER_SX_CLASS], containerWrapperStyles]}
        testID={containerWrapperTestId}
        ref={ref}>
        <View
          style={[containerGeneratedStyles[BOX_CLASSNAMES.RN_NIX_CONTAINER_INNER_SX_CLASS], style]}
          testID={testID}
          {...props}>
          {children}
        </View>
      </View>
    );
  },
);
