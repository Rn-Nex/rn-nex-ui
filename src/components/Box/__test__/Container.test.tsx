import { render } from '@testing-library/react-native';
import React from 'react';
import { StyleProp, Text, View, ViewStyle } from 'react-native';
import { Container } from '../Container';
import { LG_MAX_WIDTH, MD_MAX_WIDTH, SM_MAX_WIDTH, XS_MAX_WIDTH } from '../constants';

describe('Container Component', () => {
  const mockRef = React.createRef<View>();
  const containerMockTestId = 'mock_container';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<Container />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should match the snapshot with sx styles', () => {
    const sx = { bg: 'red', p: 10, r: 5, mt: 10 };
    const { toJSON } = render(
      <Container sx={sx}>
        <View />
      </Container>,
    );

    expect(toJSON).toMatchSnapshot();
  });

  it('renders correctly with default props', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId}>
        <Text testID="child">Child Content</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container).toBeTruthy();
  });

  it('applies custom styles (sx prop) correctly', () => {
    const { getByTestId } = render(
      <Container sx={{ bg: 'blue', p: 20, w: '90%' }} testID={containerMockTestId}>
        <Text>Test</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(
      expect.arrayContaining([{ backgroundColor: 'blue', padding: 20, margin: 'auto', width: '90%' }]),
    );
  });

  it('should remove the padding from the container when disabled disableGutters prop passed', () => {
    const { getByTestId } = render(<Container disableGutters testID={containerMockTestId} />);

    const expectedStyles: StyleProp<ViewStyle> = [{ margin: 'auto', width: '100%' }];

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(expect.arrayContaining(expectedStyles));
  });

  it('should applies maxWidth("md") prop correctly', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId} maxWidth="md" disableGutters>
        <Text>Test</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(expect.arrayContaining([{ margin: 'auto', width: MD_MAX_WIDTH }]));
  });

  it('applies maxWidth("xs") and disableGutters correctly', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId} maxWidth="xs" disableGutters>
        <Text>Test</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(expect.arrayContaining([{ margin: 'auto', width: XS_MAX_WIDTH }]));
  });

  it('should applies maxWidth("lg") prop correctly', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId} maxWidth="lg" disableGutters>
        <Text>Test</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(expect.arrayContaining([{ margin: 'auto', width: LG_MAX_WIDTH }]));
  });

  it('applies maxWidth("sm") and disableGutters correctly', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId} maxWidth="sm" disableGutters>
        <Text>Test</Text>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    expect(container.props.style).toEqual(expect.arrayContaining([{ margin: 'auto', width: SM_MAX_WIDTH }]));
  });

  it('applies container wrapper styles correctly', () => {
    const { getByTestId } = render(
      <Container testID={containerMockTestId} containerStyles={{ padding: 10 }}>
        <Text>Test</Text>
      </Container>,
    );

    const wrapper = getByTestId(containerMockTestId);

    const expectedStyles: StyleProp<ViewStyle> = [{ margin: 'auto', width: '100%', padding: 10 }];

    expect(wrapper.props.style).toEqual(expect.arrayContaining(expectedStyles));
  });

  it('renders children components correctly', () => {
    const { getByText } = render(
      <Container>
        <Text>Child Component</Text>
      </Container>,
    );

    expect(getByText('Child Component')).toBeTruthy();
  });

  it('should render dynamic styles correctly', () => {
    const { getByTestId } = render(<Container testID={containerMockTestId} containerStyles={{ backgroundColor: 'red' }} />);

    const containerWrapperTestId = `${containerMockTestId}_outer`;
    const container = getByTestId(containerWrapperTestId);

    expect(container.props.style).toEqual(expect.arrayContaining([{ width: '100%' }, { backgroundColor: 'red' }]));
  });

  it('should render the inner component with forward ref', () => {
    const innerComponentTestId = containerMockTestId + '_inner_view_test_id';

    render(
      <Container>
        <View ref={mockRef} testID={innerComponentTestId} />
      </Container>,
    );

    expect(mockRef.current).not.toBeNull();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should apply the center alignment styles to the child component', () => {
    const { getByTestId } = render(
      <Container
        disableGutters
        testID={containerMockTestId}
        sx={{ d: 'flex', fDirection: 'column', items: 'center', content: 'center' }}>
        <View>Test</View>
      </Container>,
    );

    const container = getByTestId(containerMockTestId);
    const expectedStyles: StyleProp<ViewStyle> = [
      { display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center', margin: 'auto', width: '100%' },
    ];

    expect(container.props.style).toEqual(expect.arrayContaining(expectedStyles));
  });

  it('should forward the ref correctly', () => {
    render(<Container ref={mockRef} />);

    expect(mockRef.current).not.toBeNull();
    expect(mockRef.current).toBeInstanceOf(View);
  });
});
