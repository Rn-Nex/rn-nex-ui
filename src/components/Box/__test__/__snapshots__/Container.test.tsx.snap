// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should match the snapshot 1`] = `[Function]`;

exports[`Container Component should match the snapshot with sx styles 1`] = `[Function]`;
